lib/slack-ruby-bot/commands/base.rb

```diff
@@ -122,6 +131,23 @@ def finalize_routes!
           command command_name_from_class
         end
 
+        def match_attachments(data, route)
+          fields_to_scan = %i[pretext text title]
+          data.attachments.each do |attachment|
+            fields_to_scan.each do |field|
+              next unless attachment[field]
+              match = route.match(attachment[field])
+              return match if match
+            end
+          end
+          false
+        end
+
+        def data_missing?(match_method, expression, data)
+          return true if match_method != :attachment && expression.nil?
+          return true if match_method == :attachment && (data.attachments.nil? || data.attachments.empty?)
```

Comment:
I feel like this muddles two somewhat separate things and isn't reused elsewhere. I would actually roll this method into two `next` statements above without extracting into a method or write this here as a `case` statement. Either way return `false` explicitly (I think currently it's `nil`).

Reply:
I'll take a look on this, thanks.